packages/adblocker-puppeteer/adblocker.ts

```diff
@@ -289,32 +294,38 @@ export class PuppeteerBlocker extends FiltersEngine {
       request.isMainFrame() ||
       (request.type === 'document' && frame !== null && frame.parentFrame() === null)
     ) {
-      details.continue();
+      details.continue(details.continueRequestOverrides(), this.priority);
```

Comment:
Very sorry to meddle in the comments of a merged PR. If your handler has no opinion on further action and the intent is to bypass/defer to other handlers in cooperative mode, use `details.continue(details.continueRequestOverrides(), 0)` because `0` is the de-facto default priority.

Reply:
Shouldn't this be already the case, though? The default value is `undefined` which should preserve the legacy mode (non-cooperative), then if `setRequestInterceptionPriority` is called without argument the priority defaults to `0`. And only if the user decides to use an explicit value different than `0` will `this.priority` be different from `0`. Does this seem correct?

Reply:
I think there are two cases. In one case the handler might just mean to provide a default continuation action so the request doesn't hang. In the other case, it might actually mean to override a lower-priority action.

I added a detailed discussion about this, please tell me what you think: https://github.com/puppeteer/puppeteer/pull/7796/files#diff-9eddf4dc51bf6a0125ca7fb094468ad284112270385c41cebce4f8f0a29620abR2617

Reply:
Here is an example:

Plugin A calls `.continue({},1)`
Plugin B calls `.abort('..',0)`

So we use `setRequestInterceptionPriority(2)` to make sure ads are blocked and not continued by Plugin A. But now, we have broken Plugin B because adBlocker will call `.continue({},2)`. What we really meant was `.abort('..',2)` and `,continue({},0)`.

Reply:
I see, that makes sense. Then we should probably fix this.

Reply:
See https://github.com/puppeteer/puppeteer/pull/7796/files#diff-9eddf4dc51bf6a0125ca7fb094468ad284112270385c41cebce4f8f0a29620abR2741 for a flexible solution that addresses both cases.

Reply:
`abort > respond > continue` (for the same priority). it probably makes sense to split priority for each case, e.g. continue with 0 and block with 2.

```js
interface ResolutionStrategy {
  abortPriority: number;
  continuePriority: number;
}

const DEFAULT_STRATEGY: ResolutionStrategy = {
  abortPriority: 0,
  continuePriority: 0,
};
```

Reply:
@akornatskyy Also allow for `undefined` to preserve legacy 

```ts
// This approach supports multiple priorities based on situational
// differences. You could, for example, create a config that
// allowed separate priorities for PNG vs JPG.
const DEFAULT_CONFIG: InterceptResolutionConfig = {
  abortPriority: undefined,     // Default to Legacy Mode
  continuePriority: undefined,  // Default to Legacy Mode
};

// Defaults to undefined which preserves Legacy Mode behavior
let _config: Partial<InterceptResolutionConfig> = {};

export const setInterceptResolutionConfig = (config: InterceptResolutionConfig) =>
  (_config = { ...DEFAULT_CONFIG, ...config });
```

Reply:
@akornatskyy The new methods are already released in Puppeteer v13 🎉 

In addition to the check that @benallfree recommended, I'd update the `details.continueRequestOverrides()` to `details.continueRequestOverrides?.()` as this method didn't exist before Puppeteer 10.2.0 and it currently breaks for older versions that are still supported by this package.

Reply:
@FdezRomero That's a great point, I might add that to the official docs/samples.